### PR TITLE
FEAT: Add Meilisearch in theme.

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**
Add meilisearch for theme options.

### CODE REVIEW

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/theme.conf
Comment (RobPasMue) [reply src/ansys_sphinx_theme/theme/ansys_sphinx_theme/theme.conf:15]: Yeah agreed. Let's try to be as homogeneous as possible

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/static/css/meilisearch.css
Comment (Revathyvenugopal162) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/static/css/meilisearch.css:74]:
```suggestion
    position: relative;
    padding-left: 3rem
```

File: .github/workflows/ci_cd.yml
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:54]: Please revert this.
```suggestion
    if: ${{ github.ref == 'refs/heads/main' }}
```

File: doc/source/user_guide/options.rst
Comment (RobPasMue) [inline doc/source/user_guide/options.rst:150]:
```suggestion
    ``ansys-sphinx-theme`` uses the default search functionality 
```

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/theme.conf
Comment (RobPasMue) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/theme.conf:15]: Are we going to set some default values here?

**Review by RobPasMue — APPROVED** [review]
LGTM - some minor comments

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/theme.conf
Comment (Revathyvenugopal162) [reply src/ansys_sphinx_theme/theme/ansys_sphinx_theme/theme.conf:15]: Since we are not forcing anybody to use meilisearch , i think we can leave as an empty value (if using MeiliSearch functionality, else pydata functionality) , and may add the defaults, once the libs starts using it.

Comment (RobPasMue) [reply src/ansys_sphinx_theme/theme/ansys_sphinx_theme/theme.conf:15]: Sounds good to me then. Thanks @Revathyvenugopal162

**Comment by Revathyvenugopal162** [comment]
image
Added some minor changes on the searchbar style.

**Review by akaszynski — APPROVED** [review]
Cursory LGTM after reverting the merge on main… however verify that the search box doesn’t overextend. Does on my browser when zoomed in.

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/static/css/meilisearch.css
Comment (Revathyvenugopal162) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/static/css/meilisearch.css:89]:
```suggestion
    margin-left: 41em;
```

**Comment by Revathyvenugopal162** [comment]
> Cursory LGTM after reverting the merge on main… however verify that the search box doesn’t overextend. Does on my browser when zoomed in.

Noted and Fixed.

**Review by MaxJPRey — APPROVED** [review]

**Pull request merged**
